### PR TITLE
Fix crowbar distance to work on adjacent tiles

### PR DESCRIPTION
**Pull request by rok-povsic**
This limits crowbar to work on adjacent tiles only.

I assume here that adjacent are not more than 1.5 tiles far away.

Do you propose this code to be moved to some other place? Or is there already some code that calculates this distance?

### CODE REVIEW

File: Content.Server/GameObjects/Components/Interactable/Tools/CrowbarComponent.cs [hunk 1]
```diff
@@ -18,6 +19,8 @@ public class CrowbarComponent : ToolComponent, IAfterAttack
         [Dependency] private readonly IMapManager _mapManager;
 #pragma warning restore 649
 
+        private double MAX_DISTANCE = 1.5;
```
Comment (PJB3005) [inline Content.Server/GameObjects/Components/Interactable/Tools/CrowbarComponent.cs]: You should instead use the `InteractionSystem.InteractionRange` constant to do this distance check. Then it's consistent with other item interactions.

File: Content.Server/GameObjects/Components/Interactable/Tools/CrowbarComponent.cs [hunk 2]
```diff
@@ -27,6 +30,16 @@ public void AfterAttack(AfterAttackEventArgs eventArgs)
         {
             var mapGrid = _mapManager.GetGrid(eventArgs.ClickLocation.GridID);
             var tile = mapGrid.GetTileRef(eventArgs.ClickLocation);
+
+            var ownerLocation = Owner.Transform.GridPosition;
+            var tileCenterX = tile.X + 0.5;
+            var tileCenterY = tile.Y + 0.5;
+
+            if (Distance(ownerLocation.X, tileCenterX, ownerLocation.Y, tileCenterY) > MAX_DISTANCE)
```
Comment (PJB3005) [inline Content.Server/GameObjects/Components/Interactable/Tools/CrowbarComponent.cs]: You can use `IMapGrid.GridTileToLocal` to do the work to get the `GridPosition` of the exact tile.

You can then use `GridCoordinates.Distance` to calculate the distance.

**Comment by rok-povsic** [comment]
@PJB3005 Thanks for the review, changes done.

File: Content.Server/GameObjects/Components/Interactable/Tools/CrowbarComponent.cs [hunk 3]
```diff
@@ -35,5 +45,10 @@ public void AfterAttack(AfterAttackEventArgs eventArgs)
                _entitySystemManager.GetEntitySystem<AudioSystem>().Play("/Audio/items/crowbar.ogg", Owner);
             }
         }
+
+        private double Distance(float x1, double x2, float y1, double y2)
```
Comment (PJB3005) [inline Content.Server/GameObjects/Components/Interactable/Tools/CrowbarComponent.cs]: You're not using this anymore so you can remove it.

**Pull request merged**